plugins/modules/aci_esg.py

```diff
@@ -0,0 +1,520 @@
+            ['state', 'present', ['tenant', 'ap', 'esg', 'vrf']],
+        ],
+        required_together=[['match_key', 'match_value']],
+    )
+
+    aci = ACIModule(module)
+    tenant = module.params.get('tenant')
+    ap = module.params.get('ap')
+    esg = module.params.get('esg')
+    description = module.params.get('description')
+    epg = module.params.get('epg')
+    admin_state = module.params.get('admin_state')
+    vrf = module.params.get('vrf')
+    intra_esg_isolation = module.params.get('intra_esg_isolation')
+    preferred_group_member = module.params.get('preferred_group_member')
+    state = module.params.get('state')
+    name_alias = module.params.get('name_alias')
+    epg_selector_description = module.params.get('epg_selector_description')
+    esg_contract_master = module.params.get('esg_contract_master')
+    value_operator = module.params.get('value_operator')
+    match_key = module.params.get('match_key')
+    match_value = module.params.get('match_value')
+    tag_selector_description = module.params.get('tag_selector_description')
+    subnet_selector_description = module.params.get('subnet_selector_description')
+    subnet_ip = module.params.get('subnet_ip')
+
+    if admin_state is True:
+        admin_state = 'yes'
+    else:
+        admin_state = 'no'
+
+    # VRF Selection - fvRsScope
+    child_configs = [dict(fvRsScope=dict(attributes=dict(tnFvCtxName=vrf)))]
+
+    # Tag Selector - fvTagSelector
+    if None not in (match_key, match_value) and '' not in (match_key, match_value):
+        child_configs.append(
+            dict(
+                fvTagSelector=dict(
+                    attributes=dict(
+                        matchKey=match_key,
+                        matchValue=match_value,
+                        valueOperator=value_operator,
+                        descr=tag_selector_description,
+                    )
+                )
+            )
+        )
+
+    # EPG Selector - fvEPgSelector
+    if epg is not None:
+        child_configs.append(
+            dict(
+                fvEPgSelector=dict(
+                    attributes=dict(
+                        matchEpgDn="uni/tn-{0}/ap-{1}/epg-{2}".format(tenant, ap, epg),
+                        descr=epg_selector_description,
+                    )
+                )
+            )
+        )
+
+    # IP Subnet Selector - fvEPSelector
+    if subnet_ip is not None:
+        child_configs.append(
+            dict(
+                fvEPSelector=dict(
+                    attributes=dict(
+                        matchExpression="ip=='{0}'".format(subnet_ip),
+                        descr=subnet_selector_description,
+                    )
+                )
+            )
+        )
+
+    # ESG Contract Master - fvRsSecInherited
+    if esg_contract_master is not None:
+        tDn = "uni/tn-{0}/ap-{1}/esg-{2}".format(tenant, ap, esg_contract_master)
+        child_configs.append(dict(fvRsSecInherited=dict(attributes=dict(tDn=tDn))))
```

Comment:
Selectors and EPG masters are a 1 to N relationship we create dedicated modules for 1 to N relationships so they can be better manipulated through Ansible structure of controls (loops, conditions,...)